src/MagentoAccess/MagentoService.cs

```
@@ -295,6 +315,36 @@ public MagentoService( MagentoAuthenticatedUserCredentials magentoAuthenticatedU
 			}
 		}
 
+		/// <summary>
+		///	In Magento 2.0+ we can check version via url
+		/// </summary>
+		/// <param name="mark"></param>
+		/// <returns></returns>
+		private async Task< MagentoStoreVersion > GetMagentoStoreVersionAsync( Mark mark = null )
+		{
+			mark = mark ?? Mark.CreateNew();
+			MagentoStoreVersion magentoStoreVersion = null;
+
+			try
+			{
+				using( var httpClient = new HttpClient() {  BaseAddress = new Uri( this.Credentials.BaseMagentoUrl ) } )
+				{
+					MagentoLogger.LogTraceStarted( this.CreateMethodCallInfo( mark : mark ) );
+					var storeVersionRaw = await httpClient.GetStringAsync( "magento_version" ).ConfigureAwait( false );
+					MagentoLogger.LogTraceEnded( this.CreateMethodCallInfo( mark : mark, methodResult : storeVersionRaw ) );
+
+					if ( !string.IsNullOrEmpty( storeVersionRaw ) )
+						return storeVersionRaw.ParseMagentoStoreInfoString();
+				}
+			}
+			catch ( Exception )
+			{
+				// only Magento 2.0+ version support this feature

```

Do we know what this logic will be? Maybe throw a specialized exception, so that the calling method can react appropriately?

---

I don't think that this is "exceptional" situation. Magento store owners can disable this feature and in this case we'll use the old way of determining Magento store version. Also throwing exception will complicate client code and we have to write something like this:
` try 
{
    var storeVersion = await this.GetMagentoStoreVersionAsync( mark ).ConfigureAwait( false );

   if ( storeVersion != null & storeVersion.Major >= 2 && storeVersion.Minor > 1 )
    // do something useful
    
  // throw ex to go to fallback or rewrite this method completely
   throw new FailedToDetermineMagentoStoreVersion(); 
}
catch( FailedToDetermineMagentoStoreVersion exception)
{
   // use the old way to determine Magento store version
}`

---

I understand now